src/main/java/cs2103/v15_1j/jimjim/searcher/JJSearcher.java

```diff
@@ -0,0 +1,80 @@
+package cs2103.v15_1j.jimjim.searcher;
+import java.util.List;
+
+import cs2103.v15_1j.jimjim.model.DataLists;
+import cs2103.v15_1j.jimjim.model.DeadlineTask;
+import cs2103.v15_1j.jimjim.model.Event;
+import cs2103.v15_1j.jimjim.model.FloatingTask;
+
+public class JJSearcher implements Searcher {
+	@Override
+	public DataLists search(List<Filter> filters, DataLists masterLists) {
+		DataLists resultsList = new DataLists(); // Result to be returned
+		
+		List<DeadlineTask> masterDeadlineTasksList = masterLists.getDeadlineTasksList();
+		List<FloatingTask> masterFloatingTasksList = masterLists.getFloatingTasksList();
+		List<Event> masterEventsList = masterLists.getEventsList();
+		
+		List<DeadlineTask> searchResultsDeadlineTasksList = resultsList.getDeadlineTasksList();
+		getDeadlineTasksSearchResults(filters, masterDeadlineTasksList, searchResultsDeadlineTasksList);
```

Comment:
Or you can just call `resultsList.add()` instead of having all these `get...()` functions. It will automatically add the TaskEvent to the correct list (polymorphism)